pkg/initializer/initializer.go

```diff
@@ -46,22 +54,95 @@ import (
-	isEtcdMemberPresent := false
+	var isEtcdMemberPresent bool
 	ctx := context.Background()
 	var err error
 
-	//Etcd cluster scale-up case
-	if miscellaneous.IsMultiNode(e.Logger.WithField("actor", "initializer")) {
-		m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
-		isEtcdMemberPresent, err = m.IsMemberInCluster(ctx)
-		if !isEtcdMemberPresent && err == nil {
-			retry.OnError(retry.DefaultBackoff, func(err error) bool {
-				return err != nil
-			}, func() error {
-				return m.AddMemberAsLearner(ctx)
-			})
-			// return here after adding member as no restoration or validation needed
-			return nil
+	var sc *runtime.Scheme = scheme.Scheme
+
+	//create kubectl client
+	if err := druidv1alpha1.AddToScheme(sc); err != nil {
+		fmt.Printf("failed to register scheme: %v", err)
+		os.Exit(1)
+	}
+
+	clientSet, err := miscellaneous.GetKubernetesClientSetWithSchemeOrError(sc)
+	if err != nil {
+		e.Logger.Warnf("Failed to create clientset: %v", err)
+		return nil
+	}
+
+	etcd := &druidv1alpha1.Etcd{}
+	//Read etcd
+	podName := os.Getenv("POD_NAME")
+	podNameSpace := os.Getenv("POD_NAMESPACE")
+	etcdName := podName[:strings.LastIndex(podName, "-")]
+	if err := clientSet.Get(ctx, client.ObjectKey{
+		Namespace: podNameSpace,
+		Name:      etcdName,
+	}, etcd); err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Could not fetch etcd CR %v/%v with error: %v", etcdName, podNameSpace, err),
+		}
+	}
+
+	_, bootstrapCaseFound := etcd.Annotations["druid.gardener.cloud/bootstrap"]
+	_, quorumLossCaseFound := etcd.Annotations["druid.gardener.cloud/quorum-loss"]
+
+	if !bootstrapCaseFound {
+		inputFileName := miscellaneous.EtcdConfigFilePath
+		configYML, err := ioutil.ReadFile(inputFileName)
+		if err != nil {
+			return fmt.Errorf("unable to read etcd config file: %v", err)
+		}
+
+		config := map[string]interface{}{}
+		if err := yaml.Unmarshal([]byte(configYML), &config); err != nil {
+			return fmt.Errorf("unable to unmarshal etcd config yaml file: %v", err)
+		}
+
+		initialClusterURLs := strings.Split(fmt.Sprint(config["initial-cluster"]), ",")
+
+		// set the flag FORCE_CLUSTER=true if this instance is the first pod after quorum loss
+		if quorumLossCaseFound && podName == strings.Split(initialClusterURLs[0], "=")[0] {
+			// set environment variable to force start a cluster with the member
+			os.Setenv("FORCE_CLUSTER", "true")
+		} else {
+			if etcd.Spec.Replicas > 1 {
+				m := member.NewMemberControl(e.Config.EtcdConnectionConfig)
+				isEtcdMemberPresent, err = m.IsMemberInCluster(ctx)
+				if err != nil {
+					return fmt.Errorf("this member restarted but can't be determined whether it was part of the cluster: %v", err)
+				}
+
+				if !isEtcdMemberPresent {
+					backOff := wait.Backoff{
+						Steps:    100,
+						Duration: 30 * time.Second,
+						Factor:   5.0,
+						Jitter:   0.1,
+					}
+
+					retry.OnError(backOff, func(err error) bool {
+						return err != nil
+					}, func() error {
```

Comment:
```suggestion
					retry.OnError(backOff, AnyError, func() error {
```
please see [here](https://github.com/gardener/etcd-backup-restore/blob/master/pkg/errors/error.go#L38)

Reply:
Done